src/cider/nrepl/middleware/stacktrace.clj

```diff
@@ -128,14 +128,17 @@
   "In order to match more namespaces, we look for a common namespace
   prefix across the ones we have identified."
   (let [common
-        (try (reduce
-              (fn [common ns]
-                (let [ns (str/lower-case ns)
-                      matched (map vector common ns)
-                      coincident (take-while (fn [[a b]] (= a b)) matched)]
-                  (apply str (map first coincident))))
-              (str/lower-case (first directory-namespaces))
-              directory-namespaces)
+        (try (->> directory-namespaces
+                  ;; remove usually present namespaces
+                  ;; that would break the commonality:
+                  (remove #{'dev 'user})
+                  (reduce
+                   (fn [common ns]
+                     (let [ns (str/lower-case ns)
+                           matched (map vector common ns)
```

Comment:
Maybe while I'm here I could change `map vector` (which converts a string into a vec of chars) into something else.

Else the algo doesn't seem very efficient, especially for projects with many namespaces.

Can add test coverage.

Reply:
> Maybe while I'm here I could change map vector (which converts a string into a vec of chars) into something else.

Fine by me.